patients/consumer.go

```diff
@@ -206,6 +223,86 @@ func (p *PatientCDCConsumer) sendUploadReminder(userId string) error {
 	return p.mailer.SendEmailTemplate(ctx, template)
 }
 
+func (p *PatientCDCConsumer) sendDexcomConnectEmail(userId, clinicId, clinicianId, patientName string, IsCustodial bool) error {
+	ctx, cancel := context.WithTimeout(context.Background(), defaultTimeout)
+	defer cancel()
+
+	email, err := p.getUserEmail(userId)
+	if err != nil {
+		return err
+	}
+
+	if email == "" {
+		// Abort early if the user was fetched but does not have an email address
+		return nil
+	}
+
+	clinicName, err := p.getClinicName(ctx, clinicId)
+	if err != nil {
+		return err
+	}
+
+	clinicianName, err := p.getClinicianName(ctx, clinicId, clinicianId)
+
+	restrictedTokenPaths := []string{"/v1/oauth/dexcom"}
+	restrictedTokenExpirationTime := time.Now().Add(time.Hour * 24 * 30)
+
+	// Create new or update existing restricted token for this path and user
+	currentRestrictedTokens, err := p.getUserRestrictedTokens(userId)
+	if err != nil {
+		return err
+	}
+
+	var currentRestrictedTokenId string
+	for _, token := range currentRestrictedTokens {
+		if reflect.DeepEqual(token.Paths, &restrictedTokenPaths) {
+			currentRestrictedTokenId = token.ID
+			break
+		}
+	}
+
+	var restrictedToken clients.RestrictedToken
+	if currentRestrictedTokenId != "" {
+		updatedRestrictedToken, err := p.auth.UpdateRestrictedToken(currentRestrictedTokenId, restrictedTokenExpirationTime, restrictedTokenPaths, p.shoreline.TokenProvide())
+		if err != nil {
+			return err
+		}
+		restrictedToken = *updatedRestrictedToken
+	} else {
+		createdRestrictedToken, err := p.auth.CreateRestrictedToken(userId, restrictedTokenExpirationTime, restrictedTokenPaths, p.shoreline.TokenProvide())
+		if err != nil {
+			return err
+		}
+		restrictedToken = *createdRestrictedToken
+	}
+
+	// Send the email with restricted token ID
+	p.logger.Infow("Sending Dexcom connect email",
+		"userId", userId,
+		"email", email,
+		"clinicId", clinicId,
+		"restrictedTokenId", restrictedToken.ID,
```

Comment:
I don't think the restricted token should be logged, because it is a sensitive value.

Reply:
Good catch.